### PR TITLE
Add knowledge checks and links to Introduction to RSpec lesson

### LINKED ISSUE #22930: Add missing Knowledge Checks, Learning Outcomes, and Knowledge Check links to Ruby/Introduction to RSpec
Related to #22467

This task is to go through the lesson referenced in the title of this issue and (as applicable):

- Add missing Learning Outcomes
- Add missing Knowledge Checks
- Add answer links in the Knowledge Checks

If the lesson already has Learning Outcomes or Knowledge Checks they do not need to be added to. Many lessons only need answer links added.

Knowledge check links should consist of an `<a>` tag which adds the `knowledge-check-link` class and links to the part of the lesson (or the external resource that is linked in the lesson) where the answer to the Knowledge Check can be found. Knowledge Check links should **not** point to external resources that weren't linked to as part of the lesson content or assignment. You can link to a specific line in the lesson like so:

```html
Lorem ipsum dolor sit amet, <span id='example-knowledge-check'>consectetur adipiscing elit</span>, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to the page where the span with this id lives -->
<a class='knowledge-check-link' href='#example-knowledge-check'>How do you return a value from a function?</a>
```
Or you can link to just a section of the lesson by just linking to the section heading and following [these](https://github.com/TheOdinProject/curriculum/wiki/Contributing-to-Wiki#anchor-linking) rules for determining how to format the link. Example:

```html
### Lorem ipsum dolor sit amet

sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to a heading on the page-->
<a class='knowledge-check-link' href='#lorem-ipsum-dolor-sit-amet'>How do you return a value from a function?</a>
```

### Acceptance Criteria

- [ ] Lesson referenced in the title has Learning Outcomes, Knowledge Checks, and Knowledge Check links
- [ ] Knowledge Check links all link to the proper place in the lesson, or the proper external resource that was linked to earlier in the lesson
- [ ] Knowledge Check styling should **not** change

Please do not start work on this unless you've been assigned to it. Comment on this issue and ping me @wise-king-sullyman if you would like to be assigned to it.

Additionally, we would like for contributors to only take on one of these issues at a time. Please only request assignment for (and begin work on) an issue after the Learning Outcome/Knowledge Check/Knowledge Check link issue you're currently assigned to has been approved and merged.

Please check out [this new post on our wiki](https://github.com/TheOdinProject/curriculum/wiki/Learning-Outcomes-vs-Knowledge-Checks) about Learning Outcomes and Knowledge Checks if you need to write new Learning Outcomes or Knowledge Checks for this issue.

### PR DESCRIPTION
**Pull request by rusty-reebs**
- [X] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [X] The title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 - [X] If the PR is related to an open issue, use a [relevant keyword](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) and reference it with the `#` sign and the issue number.
- [X] If changes are requested, please make the changes in a timely manner. After you submit the changes, request another review from the maintainer (top right).

#### 1.Describe the changes made and include why they are necessary or important:

Add knowledge checks and links to lesson.

#### 2. Related Issue

closes #22930

### CODE REVIEW

File: ruby_programming/testing_with_rspec/introduction_to_rspec.md [hunk 1]
```diff
@@ -244,3 +244,13 @@ This section contains helpful links to other content. It isn't required, so cons
 - Briefly look over the RSpec styling and syntax recommended by [BetterSpecs](http://www.betterspecs.org/) and read through the first six guidelines.
 - The [RSpec Cheat Sheet](https://www.dropbox.com/s/mij3d0bhi29outc/rspec_cheatsheet_attributed.pdf) should help you avoid Googling every new bit of syntax.
 - Solidify these concepts with a [shameless plug](https://medium.com/@mindovermiles262/getting-started-with-rspec-part-1-9418909f5e53) from another Odin Project contributor.
+
+### Knowledge Checks 
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class='knowledge-check-link' href='#tests-folder'>In which folder do we keep our test files?</a>
```
Comment (wise-king-sullyman) [inline ruby_programming/testing_with_rspec/introduction_to_rspec.md:251]:
```suggestion
- <a class='knowledge-check-link' href='#tests-folder'>What do you name the folder that contains your test files?</a>
```

File: ruby_programming/testing_with_rspec/introduction_to_rspec.md [hunk 2]
```diff
@@ -244,3 +244,13 @@ This section contains helpful links to other content. It isn't required, so cons
...
+- <a class='knowledge-check-link' href='#keyword-describe'>What does the keyword `describe` define?</a>
+- <a class='knowledge-check-link' href='#keyword-it'>What does the keyword `it` define?</a>
+- <a class='knowledge-check-link' href='#expect-clauses'>How many expect clauses can we write for one test case?</a>
```
Comment (wise-king-sullyman) [inline ruby_programming/testing_with_rspec/introduction_to_rspec.md:254]:
```suggestion
- <a class='knowledge-check-link' href='#expect-clauses'>How many expect clauses can you write for one test case?</a>
```
We want to speak in the second person in Knowledge Checks whenever it makes sense to do so.

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
Looks good! I have a couple of small suggestions.

**Comment by rusty-reebs** [comment]
Sure, no problem, thanks for your help. 🙂

**Comment by rusty-reebs** [comment]
I clicked Close Pull Request, not sure if I was supposed to do that. 😬

**Comment by wise-king-sullyman** [comment]
I assume based on your last comment that you didn't really want to close it, so I'm reopening.

**Review by wise-king-sullyman — APPROVED** [review]
Thanks for doing this!

**Comment by rusty-reebs** [comment]
No worries!

**Pull request merged**